drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -0,0 +1,984 @@
+	uint8_t otp_err = BMM350_OTP_STATUS_NO_ERROR;
+	uint8_t lsb = 0, msb = 0;
+
+	if (lsb_msb) {
+		/* Set OTP command at specified address */
+		tx_buf = BMM350_OTP_CMD_DIR_READ | (addr & BMM350_OTP_WORD_ADDR_MSK);
+		ret = bmm350_reg_write(dev, BMM350_REG_OTP_CMD_REG, tx_buf);
+		if (ret) {
+			LOG_ERR("i2c xfer failed! read addr = 0x%02x, ret = %d\n", tx_buf, ret);
+			return ret;
+		}
+
+		do {
+			/* Get OTP status */
+			ret += bmm350_reg_read(dev, BMM350_REG_OTP_STATUS_REG, &rx_buf[0], 3);
+			otp_status = rx_buf[2];
+			otp_err = BMM350_OTP_STATUS_ERROR(otp_status);
+			if (otp_err != BMM350_OTP_STATUS_NO_ERROR) {
+				break;
+			}
+		} while ((!(otp_status & BMM350_OTP_STATUS_CMD_DONE)) && (ret == BMM350_OK));
+
+		if (otp_err != BMM350_OTP_STATUS_NO_ERROR) {
+			switch (otp_err) {
+			case BMM350_OTP_STATUS_BOOT_ERR:
+				ret = BMM350_E_OTP_BOOT;
+				break;
+			case BMM350_OTP_STATUS_PAGE_RD_ERR:
+				ret = BMM350_E_OTP_PAGE_RD;
+				break;
+			case BMM350_OTP_STATUS_PAGE_PRG_ERR:
+				ret = BMM350_E_OTP_PAGE_PRG;
+				break;
+			case BMM350_OTP_STATUS_SIGN_ERR:
+				ret = BMM350_E_OTP_SIGN;
+				break;
+			case BMM350_OTP_STATUS_INV_CMD_ERR:
+				ret = BMM350_E_OTP_INV_CMD;
+				break;
+			default:
+				ret = BMM350_E_OTP_UNDEFINED;
+				break;
+			}
+			return ret;
+		}
+
+		/* Get OTP L/MSB data */
+		ret += bmm350_reg_read(dev, BMM350_REG_OTP_DATA_MSB_REG, &rx_buf[0], 3);
+		msb = rx_buf[2];
+		ret += bmm350_reg_read(dev, BMM350_REG_OTP_DATA_LSB_REG, &rx_buf[0], 3);
+		lsb = rx_buf[2];
+		*lsb_msb = ((uint16_t)(msb << 8) | lsb) & 0xFFFF;
+	}
+
+	return ret;
+}
+
+static int32_t fix_sign(uint32_t inval, int8_t number_of_bits)
+{
+	int32_t ret = 0;
+	int32_t power = 0;
+
+	switch ((enum bmm350_signed_bit)number_of_bits) {
+	case BMM350_SIGNED_8_BIT:
+		power = 128; /* 2^7 */
+		break;
+	case BMM350_SIGNED_12_BIT:
+		power = 2048; /* 2^11 */
+		break;
+	case BMM350_SIGNED_16_BIT:
+		power = 32768; /* 2^15 */
+		break;
+	case BMM350_SIGNED_21_BIT:
+		power = 1048576; /* 2^20 */
+		break;
+	case BMM350_SIGNED_24_BIT:
+		power = 8388608; /* 2^23 */
+		break;
+	default:
+		power = 0;
+		break;
+	}
+
+	ret = (int32_t)inval;
+	if (ret >= power) {
+		ret = ret - (power * 2);
+	}
+
+	return ret;
+}
+
+static void bmm350_update_mag_off_sens(struct bmm350_data *data)
+{
+	uint16_t off_x_lsb_msb, off_y_lsb_msb, off_z_lsb_msb, t_off = 0;
+	uint8_t sens_x, sens_y, sens_z, t_sens = 0;
+	uint8_t tco_x, tco_y, tco_z = 0;
+	uint8_t tcs_x, tcs_y, tcs_z = 0;
+	uint8_t cross_x_y, cross_y_x, cross_z_x, cross_z_y = 0;
+
+	off_x_lsb_msb = data->otp_data[BMM350_MAG_OFFSET_X] & 0x0FFF;
+	off_y_lsb_msb = ((data->otp_data[BMM350_MAG_OFFSET_X] & 0xF000) >> 4) +
+			(data->otp_data[BMM350_MAG_OFFSET_Y] & BMM350_LSB_MASK);
+	off_z_lsb_msb = (data->otp_data[BMM350_MAG_OFFSET_Y] & 0x0F00) +
+			(data->otp_data[BMM350_MAG_OFFSET_Z] & BMM350_LSB_MASK);
+	t_off = data->otp_data[BMM350_TEMP_OFF_SENS] & BMM350_LSB_MASK;
+
+#ifdef BMM350_USE_FIXED_POINT
```

Comment:
This seems to always be defined, can we clean this up and just remove all the `#ifdef`?